test/wasm2c/add.txt

```diff
@@ -797,25 +797,25 @@ static u32 w2c_add(Z_test_instance_t* instance, u32 w2c_p0, u32 w2c_p1) {
 }
 
 /* export: 'add' */
-u32 Z_testZ_add(Z_test_instance_t* instance, u32 w2c_p0, u32 w2c_p1) {
-  return w2c_add(instance, w2c_p0, w2c_p1);
+u32 w2c_test_add(w2c_test* instance, u32 w2c_p0, u32 w2c_p1) {
+  return w2c_test_add_0(instance, w2c_p0, w2c_p1);
 }
 
-void Z_test_instantiate(Z_test_instance_t* instance) {
+void wasm_instantiate_test(w2c_test* instance) {
   assert(wasm_rt_is_initialized());
 }
 
-void Z_test_free(Z_test_instance_t* instance) {
+void wasm_free_test(w2c_test* instance) {
 }
 
-wasm_rt_func_type_t Z_test_get_func_type(uint32_t param_count, uint32_t result_count, ...) {
+wasm_rt_func_type_t wasm_get_func_type_test(uint32_t param_count, uint32_t result_count, ...) {
```

Comment:
In general this change looks great, but I'm not totally sure about this new `wasm_` prefix.

Firstly is there some reason you don't put the `test` name right after the `prefix` like we do for other `w2c_` generated functions?

e.g.

```
wasm_instantiate_test
wasm_free_test
wasm_get_func_type_test
```

vs:

```
wasm_test_instantiate
wasm_test_free
wasm_test_get_func_type
```

The latter seems more natural to me and is more in keeping with the previous generated code.

Also, using the raw `wasm_` seems little odd, I'm not sure why exactly.  IIUC we do need to do different prefixes so that these admin function can't overlap with user functions?    How about using the full `wasm2c_` for the admin functions and the abbreviated `w2c_` for user function?

Reply:
> In general this change looks great, but I'm not totally sure about this new `wasm_` prefix.
> 
> Firstly is there some reason you don't put the `test` name right after the `prefix` like we do for other `w2c_` generated functions?

I think I was worried about, e.g., what if the module is named "rt" and exports a function named "free" or something, and then wasm2c will generate a function named "wasm_rt_free" which will conflict with the runtime.

> Also, using the raw `wasm_` seems little odd, I'm not sure why exactly. IIUC we do need to do different prefixes so that these admin function can't overlap with user functions? How about using the full `wasm2c_` for the admin functions and the abbreviated `w2c_` for user function?

Sure, I think if we use `wasm2c_` for the admin functions, we can put the module name right after as you suggest because there's no risk of colliding with "rt". Then we'd end up with three prefixes/families of functions:

1. A module's own exports/functions/types/tags would look like `w2c_[modname]_[name]`
2. "Admin" wasm2c-generated functions would look like `wasm2c_[modname]_instantiate`.
3. Runtime functions would look like `wasm_rt_allocate_memory`, etc.

Reply:
Sounds good to me!

Reply:
Done!

Reply:
Nitpick: Having a distinction between `wasm2c` and `w2c` is a bit wierd to me, but also I am not able to come up with anything better, so we could go with this :+1: 

I think clean slate, I would have suggested something like

w2c_/wasm_w2c_ --- wasm2c admin functions
wasm_rt_ --- wasm runtime functions 
wasm_func_ --- wasm module's functions

But, this would have the downside of renaming pretty much everything.

Reply:
One issue with using `wasm_func_` as a prefix is that every element type would then need its own prefix too, e.g. `wasm_global_`, `wasm_table_`, `wasm_memory_`...

I agree that having both `w2c_` and `wasm2c_` is a bit weird, but it seems like the best worst option in this case.